xcode/Subconscious/Shared/Components/Detail/MemoViewerDetailMetaSheetView.swift

```diff
@@ -14,7 +14,8 @@ struct MemoViewerDetailMetaSheetView: View {
     @Environment(\.dismiss) private var dismiss
     var state: MemoViewerDetailMetaSheetModel
     var send: (MemoViewerDetailMetaSheetAction) -> Void
-
+    var onViewAuthorProfile: () -> Void
```

Comment:
I would like to funnel this through `send` so we don't have to call two things and pass two things, and everything goes through one path.

I think the right way to do this is:

- pass down a `ViewStore<MemoViewerDetailMetaSheetModel>`
- send up a new action `MemoViewerDetailMetaSheetAction.requestUserProfileDetail`
  - handle as a no-op in `MemoViewerDetailMetaSheetModel.update`
- Use cursor to tag into `MemoViewerDetailAction.requestUserProfileDetail` in 
- Use `onReceive` and `compactMap` to replay select actions from `Store<MemoViewerDetailMetaSheetModel>` on `notify`.
    - Map `MemoViewerDetailAction.requestUserProfileDetail` to an equivalent `MemoViewerDetailNotification`

This way only `MemoViewerDetailView` has to know about `notification`. The children are just consuming ordinary state or ViewStore.

WDYT?

Reply:
Yep, I like it. I believe I've implemented this as you explained above, let me know if I missed something.